.github/workflows/ci.yml

```diff
@@ -145,93 +124,117 @@ jobs:
 
           echo ::set-output name=labels::${LABELS}
           echo ::set-output name=version::${VERSION}
-
-          # Get licenses if no cache exists
-          if ! [ -d ./third_party_licenses ]; then
-            go get github.com/google/go-licenses && go-licenses save ./... --save_path third_party_licenses --force
-          fi
-
-          # Get dependencies
-          go get -d ./...
-          ls -la $HOME/go/pkg/mod
-          cp -r $HOME/go/pkg/mod .
-      - name: Build amd64
-        uses: docker/build-push-action@v2
-        with:
-          builder: ${{ steps.buildx.outputs.name }}
-          build-args: |
-            GO_BUILD_ARGS=${{ steps.prep.outputs.goargs }}
-          context: .
-          file: ./Dockerfile
-          platforms: linux/amd64
-          push: false
-          tags: operator-amd64:${{ steps.prep.outputs.version }}
-          labels: ${{ steps.prep.outputs.labels }}
-          outputs: type=docker,dest=/tmp/operator-amd64.tar
-      - name: Build arm64
-        uses: docker/build-push-action@v2
-        if: ${{ contains(github.ref, 'refs/tags/v') || contains(github.ref, 'refs/heads/master') || contains(github.ref, 'refs/heads/release-') }}
-        with:
-          builder: ${{ steps.buildx.outputs.name }}
-          build-args: |
-            GO_BUILD_ARGS=${{ steps.prep.outputs.goargs }}
-          context: .
-          file: ./Dockerfile
-          platforms: linux/arm64
-          push: false
-          tags: operator-arm64:${{ steps.prep.outputs.version }}
-          labels: ${{ steps.prep.outputs.labels }}
-          outputs: type=docker,dest=/tmp/operator-arm64.tar
-      - name: Upload artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: operator
-          path: /tmp/operator*.tar
-          retention-days: 1
+      - name: prepare properties based on env variables
+        id: prepenv
+        run: |
+          # Set output parameters.
+          echo ::set-output name=goversion::${{ env.GO_VERSION }}
```

Comment:
do we need this? aren't the env variables globally defined?

Reply:
There is an error reported if env var is used in a reusable workflow call ("unknown env context".. I don't remember exact message). Environment variable can be used directly in a "step".

```
build-amd64:
  uses: ./.github/workflows/build-dockerimage.yml
  with:
    goversion: ${{ env.GO_VERSION }}
```